Framework/Geometry/src/Instrument.cpp

```diff
@@ -1235,5 +1237,15 @@ Instrument::ContainsState Instrument::containsRectDetectors() const {
 
 } // containsRectDetectors
 
+/// Only for use by ExperimentInfo. Returns the pointer to the DetectorInfo.
+const Beamline::DetectorInfo *Instrument::detectorInfo() const {
+  return m_detectorInfo;
+}
+
+/// Only for use by ExperimentInfo. Sets the pointer to the DetectorInfo.
+void Instrument::setDetectorInfo(const Beamline::DetectorInfo *detectorInfo) {
+  m_detectorInfo = detectorInfo;
```

Comment:
I think making it a `shared_ptr` would not be a big overhead. I'll have a look if that would work.